common/views/components/Layout/index.tsx

```diff
@@ -1,39 +1,38 @@
 import { FunctionComponent, PropsWithChildren } from 'react';
 
-import { grid, SizeMap } from '@weco/common/utils/classnames';
 import { Container } from '@weco/common/views/components/styled/Container';
-
-const gridSize6 = () => ({
-  s: 12,
-  m: 6,
-  shiftM: 1,
-  l: 6,
-  shiftL: 2,
-  xl: 6,
-  shiftXl: 2,
+import {
+  GridCell,
+  SizeMap,
+} from '@weco/common/views/components/styled/GridCell';
+
+const gridSize6 = (): SizeMap => ({
+  s: [12],
+  m: [6, 2],
+  l: [6, 2],
+  xl: [6, 3],
 });
 
-const gridSize8 = (shift = true) => ({
-  s: 12,
-  m: 10,
-  shiftM: shift ? 1 : 0,
-  l: 8,
-  shiftL: shift ? 2 : 0,
-  xl: 8,
-  shiftXL: shift ? 2 : 0,
+const gridSize8 = (shift = true): SizeMap => ({
+  s: [12],
+  m: [10, shift ? 2 : 1],
+  l: [8, shift ? 3 : 1],
+  xl: [8, shift ? 3 : 1],
 });
 
-const gridSize10 = (isCentered = true) => ({
-  s: 12,
-  m: 10,
-  shiftM: isCentered ? 1 : 0,
-  l: 10,
-  shiftL: isCentered ? 1 : 0,
-  xl: 10,
-  shiftXl: isCentered ? 1 : 0,
+const gridSize10 = (isCentered = true): SizeMap => ({
+  s: [12],
+  m: [10, isCentered ? 2 : 1],
+  l: [10, isCentered ? 2 : 1],
+  xl: [10, isCentered ? 2 : 1],
 });
 
-const gridSize12 = () => ({ s: 12, m: 12, l: 12, xl: 12 });
+const gridSize12 = (): SizeMap => ({
+  s: [12],
+  m: [12],
+  l: [12],
+  xl: [12],
+});
```

Comment:
Should we move these to GridFactory? I feel like we're recreating them a few times maybe?